flare/otf.py

```diff
@@ -281,7 +288,6 @@ def run(self):
             # TODO: Reinstate velocity rescaling.
             self.md_step()
             self.curr_step += 1
-            self.checkpoint()
```

Comment:
This checkpoint was set to solve the user's issue that some OTF training could be interrupted due to external reason, or the interruption during DFT calculation. So the OTF training might not be finished. Then this checkpoint can help user to resume training from that point. Maybe we can set it to be:
```
if write_model == 3:
    self.checkpoint()
```

Reply:
I think lines 294-296 take care of this:
```
        if self.write_model >= 1:
            self.write_gp()
            self.checkpoint()
```
Including an additional call to self.checkpoint seems redundant.

Reply:
That is after the entire OTF is done, then write the state to file. But sometimes the OTF is not complete, but just get interrupted at some step, so maybe better to dump checkpoint each step (or every n steps)

Reply:
Oh right, I see now. I addressed this in fe00a32